### PR TITLE
DOCS-4289 OAuth 2.0 Support in HTTP Requests Doc Update

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Updates the Advanced Options section in API, browser, and multistep API test pages and refreshes two guides on Authentication.

### Motivation

DOCS-4289

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/synthetics/api_tests/http_tests.md [hunk 1]
```diff
@@ -55,21 +55,16 @@ After choosing to create an `HTTP` test, define your test's request.
...
-   * **Client certificate**: Authenticate through mTLS by uploading your client certificate (`.crt`) and the associated private key (`.key`) in `PEM` format.
-
-     <br/> 
+     For “Single Chunk” transfer requests to AWS S3 buckets, add `x-amz-content-sha256` containing the sha256-encoded body of the request as a header (for an empty body: `x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
+   
+   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret if you selected **Resource Owner Password**).
```
Comment (cswatt) [inline content/en/synthetics/api_tests/http_tests.md:66]:
```suggestion
   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret, or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header, or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret, if you selected **Resource Owner Password**).  
```

File: content/en/synthetics/browser_tests/actions.md [hunk 2]
```diff
@@ -355,6 +355,7 @@ To define your HTTP request:
    * **NTLM**: Add NTLM authentication credentials. Support both NTLMv2 and NTLMv1.
    * **AWS Signature v4**: Enter your Access Key ID and Secret Access Key. Datadog generates the signature for your request. This option uses the basic implementation of SigV4. Specific signatures such as AWS S3 are not supported out-of-the box.  
    For “Single Chunk” transfer requests to AWS S3 buckets, add `x-amz-content-sha256` containing the sha256-encoded body of the request as a header (for an empty body: `x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
+   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret if you selected **Resource Owner Password**).
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md:358]:
```suggestion
   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret, or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header, or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret, if you selected **Resource Owner Password**).
```

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
-  - link: 'https://www.datadoghq.com/blog/test-creation-best-practices/'
-    tag: 'Blog'
-    text: 'Best practices for creating end-to-end tests'
+    text: 'Learn about browser test steps'
 ---
 
 ## Overview 
 
-You may need to monitor journeys located behind a login. There are two ways to ensure that your Datadog Browser tests can go through the login steps of your application to perform validation on post login pages:
+<div class="alert alert-info"> Are you interested in testing applications sitting behind MFA? Visit the <a href="/synthetics/guide/app-that-requires-login/#multi-factor-authentication" target="_blank">Multi-factor authentication section</a> and <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">send us feedback</a> to help us work on the systems that matter the most to your teams.</div>
```
Comment (cswatt) [inline content/en/synthetics/guide/app-that-requires-login.md:22]:
```suggestion
<div class="alert alert-info">If you are interested in testing applications sitting behind MFA, visit the <a href="/synthetics/guide/app-that-requires-login/#multi-factor-authentication" target="_blank">Multi-factor authentication section</a> and <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">send feedback</a> to help Datadog work on the systems that matter the most to your teams.</div>
```

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 4]
```diff
@@ -65,29 +66,33 @@ Depending on the type of MFA leveraged by your application, [JavaScript steps][5
 
 <div class="alert alert-info">Datadog is constantly adding features to help you record test scenarios more easily. <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">Sending feedback</a> about the MFA systems that matter the most to you is appreciated.</div>
 
-## Leverage browser test configuration options
+## Leverage test configuration options
 
-The second way to ensure that your Datadog Browser tests can login into your applications is to leverage one or several of the available browser test configurations. You can indeed decide to apply:
+The second method is to use one or several of the available advanced configuration options for [HTTP tests][6], [HTTP steps in browser tests][7], and [multistep API tests][8].
```
Comment (cswatt) [inline content/en/synthetics/guide/app-that-requires-login.md:71]:
```suggestion
The second method is to use one or several of the available advanced configuration options for [HTTP tests][6], [HTTP steps in browser tests][7], or [multistep API tests][8]. 
```

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 5]
```diff
@@ -65,29 +66,33 @@ Depending on the type of MFA leveraged by your application, [JavaScript steps][5
...
-The second way to ensure that your Datadog Browser tests can login into your applications is to leverage one or several of the available browser test configurations. You can indeed decide to apply:
+The second method is to use one or several of the available advanced configuration options for [HTTP tests][6], [HTTP steps in browser tests][7], and [multistep API tests][8]. 
 
-- Specific headers
+You can apply the following options to your HTTP request:
+
+- Specific request headers
 - Cookies
-- Basic Auth, Digest Auth, or NTLM credentials
+- Credentials for Basic Auth, Digest Auth, NTLM credentials, AWS Signature, or OAuth 2.0
```
Comment (cswatt) [inline content/en/synthetics/guide/app-that-requires-login.md:77]:
```suggestion
- Credentials for basic authentication, digest authentication, NTLM, AWS Signature, or OAuth 2.0
```

File: content/en/synthetics/multistep.md [hunk 6]
```diff
@@ -83,6 +83,7 @@ By default, you can create up to 10 test steps. To increase this limit, contact
    * **NTLM**: Add NTLM authentication credentials. Support both NTLMv2 and NTLMv1.
    * **AWS Signature v4**: Enter your Access Key ID and Secret Access Key. Datadog generates the signature for your request. This option uses the basic implementation of SigV4. Specific signatures such as AWS S3 are not supported out-of-the box.  
    For “Single Chunk” transfer requests to AWS S3 buckets, add `x-amz-content-sha256` containing the sha256-encoded body of the request as a header (for an empty body: `x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
+   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret if you selected **Resource Owner Password**).
```
Comment (cswatt) [inline content/en/synthetics/multistep.md:86]:
```suggestion
   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret, or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header, or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret, if you selected **Resource Owner Password**).
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/synthetics/api_tests/http_tests.md [hunk 7]
```diff
@@ -55,21 +55,16 @@ After choosing to create an `HTTP` test, define your test's request.
 
    {{% tab "Authentication" %}}
 
+   Upload a private key and client certificate, select an authentication method, and provide additional configuration details as prompted.
```
Comment (HugoPct) [inline content/en/synthetics/api_tests/http_tests.md:58]: Can we keep the wording for client certificates while  moving it at the top please
https://a.cl.ly/8Lu1e6LB

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 8]
```diff
@@ -65,29 +66,33 @@ Depending on the type of MFA leveraged by your application, [JavaScript steps][5
...
-## Account security
+## Secure your account
 
 ### Secure your authentication data
 
-Store your credentials as [global variables][6] (for example, one global variable for username, another one for password) and  set these variables as secure to obfuscate their values from anyone else who has access to your instance of Datadog.
+Store your credentials as [global variables][9] (for example, one global variable for username, another one for password) and set these variables as secure to obfuscate their values from anyone else who has access to your instance of Datadog.
 
-Once you create the secure variables, you can then [import these global variables][7] into your browser tests and leverage them for your login steps.
+Once you have created secure variables, you can [import these global variables][10] into your browser tests and leverage them for your login steps, or use secure variables in your HTTP and multistep API tests.
```
Comment (HugoPct) [inline content/en/synthetics/guide/app-that-requires-login.md:93]: we do not say 'secure variables' anymore but 'hidden' or 'obfuscated' variables

**Review by HugoPct — COMMENTED** [review]
added a comment

File: content/en/synthetics/api_tests/http_tests.md [hunk 9]
```diff
@@ -55,21 +55,21 @@ After choosing to create an `HTTP` test, define your test's request.
...
       openssl pkcs12 -in <CERT>.p12 -out <CERT_KEY>.key -nodes -nocerts
       openssl pkcs12 -in <CERT>.p12 -out <CERT>.cert -nokeys
       ```
+      
+   * **HTTP Basic Auth**: Add HTTP basic authentication credentials.
+   * **Digest Auth**: Add Digest authentication credentials. 
+   * **NTLM**: Add NTLM authentication credentials. Support both NTLMv2 and NTLMv1.
+   * **AWS Signature v4**: Enter your Access Key ID and Secret Access Key. Datadog generates the signature for your request. This option uses the basic implementation of SigV4. Specific signatures such as AWS S3 are not supported out-of-the box.  
+     For “Single Chunk” transfer requests to AWS S3 buckets, add `x-amz-content-sha256` containing the sha256-encoded body of the request as a header (for an empty body: `x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
+
```
Comment (HugoPct) [inline content/en/synthetics/api_tests/http_tests.md:70]:
```suggestion
```

**Review by cswatt — APPROVED** [review]

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 10]
```diff
@@ -1,31 +1,34 @@
...
+<div class="alert alert-info">If you are interested in testing applications sitting behind MFA, visit the <a href="/synthetics/guide/app-that-requires-login/#multi-factor-authentication" target="_blank">Multi-factor authentication section</a> and <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">send feedback</a> to help Datadog work on the systems that matter the most to your teams.</div>
+
+You may need to monitor user journeys located behind a login. There are two ways to ensure that your Datadog browser tests can go through the login steps of your application to perform validation on post-login pages:
 
-- [Include the login steps in your recording](#include-the-login-steps-in-your-recording)
-- [Leverage browser test configuration options](#leverage-browser-test-configuration-options)
+- [Include the login steps in your browser test recording](#include-the-login-steps-in-your-recording)
+- [Leverage advanced options for API, multistep, and browser tests](#leverage-test-configuration-options)
```
Comment (alai97) [inline content/en/synthetics/guide/app-that-requires-login.md:27]:
```suggestion
- [Leverage advanced options in your browser tests](#leverage-test-configuration-options)
```

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 11]
```diff
@@ -1,31 +1,34 @@
...
 
-You can also ensure your credentials are securely stored and obfuscated across the application [using obfuscated global variables](#account-security).
+You can also ensure your credentials are securely stored and obfuscated across the application by [using obfuscated global variables](#account-security).
 
-<div class="alert alert-info"> Are you interested in testing applications sitting behind MFA? Visit the <a href="/synthetics/guide/app-that-requires-login/#multi-factor-authentication" target="_blank">below section</a> and <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">send us feedback</a> to help us work on the systems that matter the most to your teams.</div>
+To ensure your credentials are securely stored and obfuscated across the application, use [secured global variables](#account-security).
```
Comment (alai97) [inline content/en/synthetics/guide/app-that-requires-login.md:31]:
```suggestion
To ensure your credentials are securely stored and obfuscated across the application, use [obfuscated global variables](#account-security).
```

File: content/en/synthetics/guide/app-that-requires-login.md [hunk 12]
```diff
@@ -65,29 +68,33 @@ Depending on the type of MFA leveraged by your application, [JavaScript steps][5
...
+{{< img src="synthetics/guide/app_that_requires_login/http_request_advanced_options_oauth.png" alt="Login to your app with browser test configuration options such as OAuth 2.0" style="width:90%" >}}
 
-You can manually apply these configured headers, cookies, and credentials on the page you are recording from and then record steps your test performs post-login. By default, the browser test automatically passes through authentication with your specified headers, cookies, and/or credentials at execution time and then goes through all recorded steps.
+You can manually apply these configured headers, cookies, and credentials on the page you are recording from and record the steps your test performs post-login. 
 
-{{< img src="synthetics/guide/app_that_requires_login/bt_adv_options.jpg" alt="Login to your app with browser test configuration options">}}
+By default, the browser, API, or multistep API test automatically passes through authentication with your specified headers, cookies, and credentials at execution time and then completes all recorded steps.
```
Comment (alai97) [inline content/en/synthetics/guide/app-that-requires-login.md:87]:
```suggestion
By default, the browser test automatically passes through authentication with your specified headers, cookies, and credentials at execution time and then completes all recorded steps.
```

File: content/en/synthetics/guide/browser-tests-totp.md [hunk 13]
```diff
@@ -35,19 +35,23 @@ RBAC restrict access to global variables is in beta. To request access, contact
 
 {{< img src="synthetics/guide/browser-tests-totp/new-variable-totp.png" alt="Create a MFA token" style="width:100%;" >}}
 
-## TOTP in Synthetic tests
-You can use the secret key or QR code stored in a global variable across all your Synthetic tests. When creating a browser or API test, inject the TOTP generated from the secret key or QR code stored in the global variable to verify your application’s authentication workflow.
+## Use TOTP in your Synthetic tests
+You can use the secret key or QR code stored in a global variable across all your Synthetic tests. When creating a [browser][2], [API][3], or [multistep API test][4], inject the TOTP generated from the secret key or QR code stored in the global variable to verify your application’s authentication workflow.
```
Comment (alai97) [inline content/en/synthetics/guide/browser-tests-totp.md:39]:
```suggestion
You can use the secret key or QR code stored in a global variable across all your Synthetic tests. When creating a [browser test][2], inject the TOTP generated from the secret key or QR code stored in the global variable to verify your application’s authentication workflow.
```

File: content/en/synthetics/guide/browser-tests-totp.md [hunk 14]
```diff
@@ -35,19 +35,23 @@ RBAC restrict access to global variables is in beta. To request access, contact
...
-{{< img src="synthetics/guide/browser-tests-totp/mfa-token-totp.mp4" alt="Recording a TOTP validation" video="true" >}}
-
 ## Further Reading
 
 {{< partial name="whats-next/whats-next.html" >}}
 
 [1]: /account_management/rbac/?tab=datadogapplication#custom-roles
+[2]: /synthetics/browser_tests/
+[3]: /synthetics/api_tests/
+[4]: /synthetics/multistep/
```
Comment (alai97) [inline content/en/synthetics/guide/browser-tests-totp.md:57]:
```suggestion
```

**Review by HugoPct — APPROVED** [review]

File: content/en/synthetics/browser_tests/actions.md [hunk 15]
```diff
@@ -355,6 +355,7 @@ To define your HTTP request:
    * **NTLM**: Add NTLM authentication credentials. Support both NTLMv2 and NTLMv1.
    * **AWS Signature v4**: Enter your Access Key ID and Secret Access Key. Datadog generates the signature for your request. This option uses the basic implementation of SigV4. Specific signatures such as AWS S3 are not supported out-of-the box.  
    For “Single Chunk” transfer requests to AWS S3 buckets, add `x-amz-content-sha256` containing the sha256-encoded body of the request as a header (for an empty body: `x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
+   * **OAuth 2.0**: Choose between granting client credentials or a resource owner password and enter an access token URL. Depending on your selection, enter a client ID and secret, or a username and password. From the dropdown menu, select an option to either send the API token as a basic authentication header, or send the client credentials in the body. Optionally, you can provide additional information such as the audience, resource, and scope (as well as the client ID and secret, if you selected **Resource Owner Password**).
```
Comment (alai97) [inline content/en/synthetics/browser_tests/actions.md:358]:
```suggestion
```

**Review by HugoPct — APPROVED** [review]

**Pull request merged**